### PR TITLE
Add 'raises' to run_tool, which raises exceptions instead of logging them.

### PR DESCRIPTION
**Pull request by nbiederbeck**
With this, we can test that tools fails with the expected Exceptions.

### CODE REVIEW

File: ctapipe/core/tests/test_tool.py [hunk 1]
```diff
@@ -340,3 +342,28 @@ class MyTool(Tool):
...
+        name = "Bad"
+        description = "This tool raises an exception."
+
+        def start(self):
+            raise ValueError("1 does not equal 0.")
+
+    assert run_tool(ToolGood(), raises=True) == 0
+
+    with pytest.raises(ValueError):
+        ret = run_tool(ToolBad(), raises=True)
```
Comment (maxnoe) [inline ctapipe/core/tests/test_tool.py:365]: Add test that with `raises=False` we get a non-zero exit code for the bad tool

Comment (nbiederbeck) [reply ctapipe/core/tests/test_tool.py:367]: done

**Review by maxnoe — APPROVED** [review]

File: ctapipe/core/tests/test_tool.py [hunk 2]
```diff
@@ -340,3 +342,30 @@ class MyTool(Tool):
...
+    assert run_tool(ToolGood(), raises=True) == 0
+
+    assert run_tool(ToolBad(), raises=False) == 1
+
+    with pytest.raises(ValueError):
+        ret = run_tool(ToolBad(), raises=True)
+
+    # `ret` does not exist, because `run_tool` raised an Exception
+    with pytest.raises(UnboundLocalError):
+        assert ret == 1
```
Comment (maxnoe) [inline ctapipe/core/tests/test_tool.py:371]: `assert "ret" not in locals()`?

Comment (maxnoe) [reply ctapipe/core/tests/test_tool.py:371]: Or just leave it out, there is no possibility of ret being set when the ValueError does not happen

Comment (nbiederbeck) [reply ctapipe/core/tests/test_tool.py:371]: > there is no possibility of ret being set when the ValueError does happen

Yes, and I wanted to make that very very clear. But I can remove it

File: ctapipe/core/tool.py [hunk 3]
```diff
@@ -357,6 +358,8 @@ def run(self, argv=None):
             Provenance().finish_activity(activity_name=self.name, status="interrupted")
             exit_status = 130  # Script terminated by Control-C
         except Exception as err:
+            if raises:
```
Comment (LukasNickel) [inline ctapipe/core/tool.py:361]: Do we want this to happen before finishing the provenance log?
I realize that this is mostly meant for testing anyway, but would personally still write out the logs

Comment (maxnoe) [reply ctapipe/core/tool.py:361]: Good point.

Comment (maxnoe) [reply ctapipe/core/tool.py:361]: Actually, the provenance is being written (it is in the `finallly` block below, which always gets run).

But it is missing now the cause for the finishing of the activity. So we should definitely move the raise below the `finish_activity`.

File: ctapipe/core/tests/test_tool.py (hunk 2 above)
Comment (nbiederbeck) [reply ctapipe/core/tests/test_tool.py]: I removed it

File: ctapipe/core/tool.py (hunk 3 above)
Comment (nbiederbeck) [reply ctapipe/core/tool.py]: Done

**Review by LukasNickel — APPROVED** [review]

**Review by maxnoe — APPROVED** [review]
